src/Symfony/Component/Lock/Key.php

```diff
@@ -70,4 +71,29 @@ public function getState($stateKey)
     {
         return $this->state[$stateKey];
     }
+
+    /**
+     * @param float $ttl The expiration delay of locks in seconds.
+     */
+    public function reduceLifetime($ttl)
+    {
+        $newExpiringDate = \DateTimeImmutable::createFromFormat('U.u', (string) (microtime(true) + $ttl));
+
+        if (null === $this->expiringDate || $newExpiringDate < $this->expiringDate) {
```

Comment:
I am not sure if it really is expected behaviour that the method silently ignores any argument if it is no smaller than the existing expiration date.

Reply:
IMO it's not an issue if a store ask for a TTL greater than the current one.